decidim-core/lib/decidim/search_resource_fields_mapper.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Decidim
+  # A class with the responsibility to mapp fields between a Searchalbe and a SearchableRsrc.
```

Comment:
Typo in `Searchalbe`

Reply:
Also, this class looks rather complicated, can you add docs to both the methods and the class definition to understand why it's used, and how?